modules/iothubdeviceping/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+cmake_minimum_required(VERSION 2.8.11)
+#this is CMakeLists for the iothubdeviceping module
+
+set(iothubdeviceping_sources
+    ./src/iothubdeviceping.c
+)
+
+set(iothubdeviceping_headers
+    ./inc/iothubdeviceping.h
+)
+
+set(iothubdeviceping_hl_sources
+    ./src/iothubdeviceping_hl.c
+)
+
+set(iothubdeviceping_hl_headers
+    ./inc/iothubdeviceping_hl.h
+)
+
+include_directories(./inc)
+include_directories(${GW_INC})
+include_directories(${IOTHUB_CLIENT_INC_FOLDER} ${UAMQP_INC_FOLDER} ${SHARED_UTIL_INC_FOLDER})
+
+#this builds the IoTHubDevicePing dynamic library
+add_library(iothubdeviceping MODULE
+    ${iothubdeviceping_sources}
+    ${iothubdeviceping_headers}
+)
+target_link_libraries(iothubdeviceping
+    gateway
+    iothub_client
+)
+
+#this builds the IoTHubDevicePing static library
+add_library(iothubdeviceping_static
+    ${iothubdeviceping_sources}
+    ${iothubdeviceping_headers}
+)
+target_compile_definitions(iothubdeviceping_static PRIVATE BUILD_MODULE_TYPE_STATIC)
+target_link_libraries(iothubdeviceping_static
+    gateway
+    iothub_client
+)
+
+#this builds the IoTHubDevicePing_HL dynamic library (by default it uses IoTHub Module linked statically)
+add_library(iothubdeviceping_hl MODULE
```

Comment:
We've recently made a breaking change that will affect you here...if you want your module to support a version of the `Module_Create` function that takes a JSON string instead of a struct, you no longer create a "high-level" (HL) module to do it. Instead, we added a new function to the module interface: `Module_CreateFromJson`. To migrate to the latest SDK, you'd do the following:
- remove the `iothubdeviceping_hl` module
- Implement the `Module_CreateFromJson` function in `iothubdeviceping` (use the code from `Module_Create` in your HL module)

See any of the modules in our latest `develop` branch for examples of this.